### ISSUE TITLE
Security Issue

### ISSUE THREAD

**Issue by 7h3h4ckv157** [GoogleCloudPlatform/python-docs-samples#8315]
# Summary

As this repository is considered as Samples - (code for educational purposes), I don't think it's vital to report/submit more pieces of information to GoogleVRP regarding this repository. So I'm unveiling freely by making an issue here. 

# Details

 I reviewed the source code of your open-source project and identified a potential security flaw in your code. 

Vulnerable code link ↓

[Click Here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/476270001a9b68f4446e6b3c07ed33566a7396bd/appengine/standard_python3/cloud_debugger/main.py)

[![IMG-20220902-214912.jpg](https://i.postimg.cc/BvLW1jYy/IMG-20220902-214912.jpg)](https://postimg.cc/p9HcwLtY)

Check the code and you can see the unsanitized input from an HTTP parameter flows into the return value of ReverseString, where it is used to render an HTML page returned to the user. This allows a malicious attacker to perform Cross-Site Scripting attack (XSS) through GET request itself. The end user’s browser has no way to know that the malicious script should not be trusted, and will execute the script. Because it thinks the script came from a trusted source. 

# Reproduction steps ↓

   
```
1.  Run the python code (Start the service)

2.  Visit the service

3.  Use reversed XSS payloads. Eg: > >)niamod.tnemucod(trela=daolno/gvs<

4.  XSS can triggered on "/reverse_string?string=" parameter

5. Done
```

Proof of Concept ↓

[![poc.png](https://i.postimg.cc/8cfgFKdv/poc.png)](https://postimg.cc/p5v64B6V)

Sanitize data input in an HTTP request before reflecting it back, ensuring all data is validated, filtered or escaped before echoing anything back to the user.

I hope it helps.

Cheers,

**Comment by 7h3h4ckv157** [comment]
For reference

- [XSS](https://owasp.org/www-community/attacks/xss/)
- [XSS]( https://portswigger.net/web-security/cross-site-scripting)

**Comment by engelke** [comment]
Thank you for your report. This is fixed in #8316 by using a Flask template, which automatically sanitizes string data, for the output page.

**Comment by Ranjul-Arumadi** [comment]
Nice finding @7h3h4ckv157 🔥
